core/src/main/java/org/stellar/anchor/sep38/Sep38Service.java

```diff
@@ -79,54 +94,116 @@ public GetPricesResponse getPrices(
+      String buyAmount,
+      String buyDeliveryMethod,
+      String countryCode)
+      throws AnchorException {
+    if (this.rateIntegration == null) {
+      throw new ServerErrorException("internal server error");
+    }
+    validateAsset("sell_", sellAssetName);
+    validateAsset("buy_", buyAssetName);
+
+    if ((sellAmount == null && buyAmount == null) || (sellAmount != null && buyAmount != null)) {
+      throw new BadRequestException("Please provide either sell_amount or buy_amount");
+    } else if (sellAmount != null) {
+      validateAmount("sell_", sellAmount);
+    } else {
+      validateAmount("buy_", buyAmount);
     }
 
+    InfoResponse.Asset sellAsset = assetMap.get(sellAssetName);
+    InfoResponse.Asset buyAsset = assetMap.get(buyAssetName);
+
+    // sellDeliveryMethod
     if (!Objects.toString(sellDeliveryMethod, "").isEmpty()) {
       if (!sellAsset.supportsSellDeliveryMethod(sellDeliveryMethod)) {
         throw new BadRequestException("Unsupported sell delivery method");
       }
     }
+
+    // buyDeliveryMethod
+    if (!Objects.toString(buyDeliveryMethod, "").isEmpty()) {
+      if (!buyAsset.supportsBuyDeliveryMethod(buyDeliveryMethod)) {
+        throw new BadRequestException("Unsupported buy delivery method");
+      }
+    }
+
+    // countryCode
+    if (!Objects.toString(countryCode, "").isEmpty()) {
+      List<String> sellCountryCodes = sellAsset.getCountryCodes();
+      List<String> buyCountryCodes = buyAsset.getCountryCodes();
+      boolean bothCountryCodesAreNull = sellCountryCodes == null && buyCountryCodes == null;
+      boolean countryCodeIsSupportedForSell =
+          sellCountryCodes != null && sellCountryCodes.contains(countryCode);
+      boolean countryCodeIsSupportedForBuy =
+          buyCountryCodes != null && buyCountryCodes.contains(countryCode);
+      if (bothCountryCodesAreNull
+          || !(countryCodeIsSupportedForSell || countryCodeIsSupportedForBuy)) {
+        throw new BadRequestException("Unsupported country code");
+      }
+    }
+
+    GetRateRequest request =
+        GetRateRequest.builder()
+            .sellAsset(sellAssetName)
+            .sellAmount(sellAmount)
+            .sellDeliveryMethod(sellDeliveryMethod)
+            .buyAsset(buyAssetName)
+            .buyAmount(buyAmount)
+            .buyDeliveryMethod(buyDeliveryMethod)
+            .countryCode(countryCode)
+            .build();
+    GetRateResponse rateResponse = this.rateIntegration.getRate(request);
+
+    // Calculate amounts: sellAmount = buyAmount*price or buyAmount = sellAmount/price
+    GetPriceResponse.GetPriceResponseBuilder builder =
+        GetPriceResponse.builder().price(rateResponse.getRate().getPrice());
+    BigDecimal bPrice = new BigDecimal(rateResponse.getRate().getPrice());
+    if (sellAmount != null) {
+      BigDecimal bSellAmount = new BigDecimal(sellAmount);
+      BigDecimal bBuyAmount = bSellAmount.divide(bPrice, 4, RoundingMode.HALF_UP);
```

Comment:
hard-coded `4` here?

Reply:
Also, deciding how to round here is tough. Personally I think we should round down, but I could be convinced otherwise.

Reply:
> hard-coded `4` here?
The hardcoded `4` is fixed in the next PR: https://github.com/stellar/java-stellar-anchor-sdk/pull/125

> Also, deciding how to round here is tough. Personally I think we should round down, but I could be convinced otherwise.
Round down makes sense to me.